torch/csrc/jit/runtime/static/memory_planner.h

```diff
@@ -199,8 +192,25 @@ class MemoryPlanner {
   // We don't have any guarantee that the model doesn't change the
   // Storage for managed tensors out from under us during execution,
   // so we have to check the StorageImpls each time we deallocate.
-  std::vector<std::pair<size_t, at::StorageImpl>>
-      managed_tensor_storage_impls_{};
+
+  // We will use placement-new to add new storages to this buffer of bytes.
+  alignas(at::StorageImpl) unsigned char* storages_buffer_;
```

Comment:
This does not do what you think it does. This will ensure that the char* POINTER is aligned to StorageImpl size in the class. But you need the malloc'ed memory to be aligned instead.

Reply:
Actually, why does the malloc'ed memory need to be aligned, I don't think that's necessary either.

Reply:
I see, it is not necessary. I didn't know what `alignas` does before, I was just copying from this and thought it was important here too: https://en.cppreference.com/w/cpp/language/new#Placement_new

But I can instead just use the one `StorageImpl* storages_` pointer and initialize it like `storages_ = reinterpret_cast<StorageImpl*>(new unsigned char[storages_capacity_ * sizeof(StorageImpl)])`, and that works just fine. (The `reinterpret_cast` is necessary because `StorageImpl::StorageImpl()` without args is not allowed.)